### PR TITLE
connect form to server and fix validation issues

### PR DESCRIPTION
**Pull request by yaay**

### CODE REVIEW

File: src/app/domain/basic-information/models/basic-information.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { Azhar } from "../../azhar/models/azhar"
+import { Club } from "../../club/models/club"
+import { District } from "../../district/models/district"
+import { EducationAdministration } from "../../eduction-administration/model/eduction-administration"
+import { Government } from "../../government/models/government"
+import { YouthCenter } from "../../youth-center/models/youth-center"
+
+export interface BasicInformation {
+    "id": number,
+    "version": number,
```
Comment (Tasneem-Gad) [inline src/app/domain/basic-information/models/basic-information.ts:12]: we do not need to send or show the version  in ui

File: src/app/domain/basic-information/models/basic-information.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "fullName": string,
+    "nationalId": string,
+    "birthDate": string,
```
Comment (Tasneem-Gad) [inline src/app/domain/basic-information/models/basic-information.ts:15]: why add a property in double quotation?

File: src/app/domain/basic-information/models/basic-information.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    "female": boolean,
+    "muslim": boolean,
+    "hasDisability": boolean,
+    "disabilityType": string | null,
+    "anotherClub": string | null,
+    "affiliateParty": AffiliateParty,
+    "affiliateClub": AffiliateClub | null,
+    "governorate": Government,
+    "district": District,
+    "request": {
```
Comment (Tasneem-Gad) [inline src/app/domain/basic-information/models/basic-information.ts:25]: why not separate in another interface

File: src/app/domain/basic-information/models/basic-information.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        "id": number | null,
+        "version": number | null,
+        "requestStatus": any,
```
Comment (Tasneem-Gad) [inline src/app/domain/basic-information/models/basic-information.ts:28]: don't use any

File: src/app/domain/basic-information/request-basic-information.repository.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    getBasicInformation(id: number): Observable<any> {
+        return this.http
+            .get(this.APIUrl + `/${id}` + '/basic-information')
+            .pipe(catchError((err) => {
+                throw new Error('Error', err.message)
+            }))
+    }
+
+    addBasicInformation(id: number, resource: any): Observable<any> {
```
Comment (Tasneem-Gad) [inline src/app/domain/basic-information/request-basic-information.repository.ts:30]: don't use any

File: src/app/domain/basic-information/request-basic-information.repository.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        let formattedResource = this.extractIds.extractIds(resource);
+        formattedResource['birthDate'] = formattedResource.dob?.year + '-' + formattedResource.dob?.month + '-' + formattedResource.dob?.day;
```
Comment (Tasneem-Gad) [inline src/app/domain/basic-information/request-basic-information.repository.ts:32]: why add this validation here?

File: src/app/pages/form-stepper/pages/main-data/main-data.component.ts [hunk 7]
```diff
@@ -93,11 +132,41 @@ export class MainDataComponent {
 }
 
   ngOnInit() {
+    this.getFormDataById();
     this.getGovs();
     this.getBirthAndGenderfromId();
     this.getAffiliateParties();
   }
 
+  disabilityTypeShow() {
```
Comment (Tasneem-Gad) [inline src/app/pages/form-stepper/pages/main-data/main-data.component.ts:141]: try naming be meaningful

File: src/app/pages/form-stepper/pages/main-data/main-data.component.ts [hunk 8]
```diff
@@ -126,21 +201,21 @@ export class MainDataComponent {
   }
 
   getEduAdmin() {
-    const govId = this.basicInfoForm.value.government?.['id'];
+    const govId = this.basicInfoForm.value.governorate?.['id'];
     this.eduAdminRepository.getEducationsAdministration(govId).subscribe((response: any) => {
```
Comment (Tasneem-Gad) [inline src/app/pages/form-stepper/pages/main-data/main-data.component.ts:205]: don't use any

File: src/app/pages/form-stepper/pages/main-data/main-data.component.ts [hunk 9]
```diff
@@ -153,7 +228,7 @@ export class MainDataComponent {
   }
 
   getClubs() {
-    const govId: number = this.basicInfoForm.value.government?.['id']!;
+    const govId: number = this.basicInfoForm.value.governorate?.['id']!;
     this.clubRepository.getClubs(govId).subscribe((response: any) => {
```
Comment (Tasneem-Gad) [inline src/app/pages/form-stepper/pages/main-data/main-data.component.ts:232]: same

File: src/app/domain/basic-information/basic-information.repository.ts [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    constructor(http: HttpClient, private extractIds: ExtractIdService) {
+        super(http)
+    }
+
+    getResourceUrl(): string {
+        return 'basic-information'
+    }
+
+    update(id: number, requestId: number, resource: any): Observable<any> {
+        let formattedResource = this.extractIds.extractIds(resource);
```
Comment (Tasneem-Gad) [inline src/app/domain/basic-information/basic-information.repository.ts:21]: why make validation here

**Review by Tasneem-Gad — CHANGES_REQUESTED** [review]

**Review by Rawan2211 — APPROVED** [review]

File: src/app/pages/form-stepper/pages/main-data/main-data.component.ts [hunk 11]
```diff
@@ -165,70 +245,107 @@ export class MainDataComponent {
...
+  resetYouthCenter() {
+    if (this.affiliateClub === 'youthCenter') {
+      this.basicInfoForm.get('youthCentre')?.reset();
+      this.youthCenters = [];
+      this.getYouthCenters();
+    }
+  }
+
+  isFormValid() {
+    // console.log('cdr')
```
Comment (Tasneem-Gad) [inline src/app/pages/form-stepper/pages/main-data/main-data.component.ts:347]: remove

**Review by Tasneem-Gad — CHANGES_REQUESTED** [review]

**Review by Rawan2211 — APPROVED** [review]

**Pull request merged**